Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -112,61 +112,75 @@ namespace Multiplayer
                 }
                 else
                 {
                     // If we're a client yet are lacking a connection then we've been unintentionally disconnected
                     // Display a disconnect message in the viewport
-                    DrawConnectionStatus(AzNetworking::ConnectionState::Disconnected);
+                    DrawConnectionStatus(AzNetworking::ConnectionState::Disconnected, m_hostIpAddress);
                 }
             }
         }
     }
 
-    void MultiplayerConnectionViewportMessageSystemComponent::DrawConnectionStatus(AzNetworking::ConnectionState connectionState)
+    void MultiplayerConnectionViewportMessageSystemComponent::DrawConnectionStatus(AzNetworking::ConnectionState connectionState, const AzNetworking::IpAddress& hostIpAddress)
     {
-        AZ::RPI::ViewportContextPtr viewport = AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext();
-        if (!viewport)
-        {
-            return;
-        }
-
-        // Draw the status (example: Connected or Disconnected)
-        const char* connectionStateText = ToString(connectionState).data();
-        const AzFramework::WindowSize viewportSize = viewport->GetViewportSize();
-        m_drawParams.m_hAlign = AzFramework::TextHorizontalAlignment::Right;
-        m_drawParams.m_position = AZ::Vector3(static_cast<float>(viewportSize.m_width), static_cast<float>(viewportSize.m_height), 1.0f) + AZ::Vector3(viewportConnectionBottomRightBorderPadding) * viewport->GetDpiScalingFactor();
-        
+        AZ::Color connectionStateColor;
         switch (connectionState)
         {
         case AzNetworking::ConnectionState::Connecting:
-            m_drawParams.m_color = AZ::Colors::Yellow;
+            connectionStateColor = AZ::Colors::Yellow;
             break;
         case AzNetworking::ConnectionState::Connected:
-            m_drawParams.m_color = AZ::Colors::Green;
+            connectionStateColor = AZ::Colors::Green;
             break;
         case AzNetworking::ConnectionState::Disconnecting:
-            m_drawParams.m_color = AZ::Colors::Orange;
+            connectionStateColor = AZ::Colors::Orange;
             break;
         case AzNetworking::ConnectionState::Disconnected:
-            m_drawParams.m_color = AZ::Colors::Red;
+            connectionStateColor = AZ::Colors::Red;
             break;
-        default: m_drawParams.m_color = AZ::Colors::White;
+        default:
+            connectionStateColor = AZ::Colors::White;
         }
 
-        m_fontDrawInterface->DrawScreenAlignedText2d(m_drawParams, connectionStateText);
+        // Draw our host's remote ip address
+        const auto multiplayerSystemComponent = AZ::Interface<IMultiplayer>::Get();
+        MultiplayerAgentType agentType = multiplayerSystemComponent->GetAgentType();
+        if (agentType == MultiplayerAgentType::Client)
+        {
+            DrawConnectionStatusLine(hostIpAddress.GetString().c_str(), connectionStateColor);
```

Comment:
at first glance it might not be apparent whether this is the server IP or our own IP address (esp. testing locally when they'd both be "127.0.0.1:something". What do you think about passing in a formatted string like "server address <IP_ADDRESS>" so it's super explicit what this is?

Reply:
Perfect, thanks :)